daml-script/test/JsonTestRunner.hs

```diff
@@ -0,0 +1,54 @@
+-- Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module JsonTestRunner (main) where
+
+import Control.Concurrent
+import DA.Bazel.Runfiles
+import DA.Daml.Helper.Run
+import DA.PortFile
+import Data.Aeson
+import qualified Data.HashMap.Strict as HashMap
+import qualified Data.Map.Strict as Map
+import qualified Data.Text as T
+import qualified Data.Text.Extended as T
+import System.FilePath
+import System.IO.Extra
+import System.Process
+import qualified Web.JWT as JWT
+
+main :: IO ()
+main = do
+    sandbox <- locateRunfiles (mainWorkspace </> "ledger" </> "sandbox" </> exe "sandbox-binary")
+    jsonApi <- locateRunfiles (mainWorkspace </> "ledger-service" </> "http-json" </> exe "http-json-binary")
+    logback <- locateRunfiles (mainWorkspace </> "ledger-service" </> "http-json" </> "release" </> "json-api-logback.xml")
+    testClient <- locateRunfiles (mainWorkspace </> "daml-script" </> "test" </> exe "test_client_json")
+    dar <- locateRunfiles (mainWorkspace </> "daml-script" </> "test" </> "script-test.dar")
+    withTempFile $ \tempFile -> do
+        withCreateProcess (proc sandbox [dar, "--port-file", tempFile, "--wall-clock-time", "--ledgerid=myledger"]) $ \_ _ _ _ -> do
+        _ <- readPortFile maxRetries tempFile
+        withCreateProcess (proc jsonApi ["--ledger-host=localhost", "--ledger-port=6865", "--http-port=7500", "--wrapper_script_flag=--jvm_flag=-Dlogback.configurationFile=" <> logback]) $ \_ _ _ _ -> do
```

Comment:
Shouldn't this read the port from the port file?

Reply:
I’ve changed it to use a random port for sandbox for now and use the one from the port file.